### DISCUSSION TITLE
Unable to open URL: chave://

### DISCUSSION THREAD

**Discussion by expend20** [streamlink/streamlink#4537] [Q&A] [answered]
Hello, stumbled upon this error:
```
[stream.hls][error] Failed to create decryptor: Unable to open URL: chave://pi1CcgormCfrBeHFonx1...
```

m3u8 file looks like this:
```
#EXTM3U
#EXT-X-VERSION:3
#EXT-X-PLAYLIST-TYPE:VOD
#EXT-X-TARGETDURATION:7
#EXT-X-MEDIA-SEQUENCE:0
#EXTINF:6.0,
#EXT-X-KEY:METHOD=AES-128,URI="chave://pi1CcgormCfrBeHFonx1SjYyNREActOy9x5fdSmTcQ5yl1p3+B2aztf1Zt/KqeWbEe9A5aYBXkkQaAo/0gXvlYb1f2QarZUd26JK+Y3rgjiuOuijkHMqtJ7B7KZryLaaUSYnG2IxTR6Ijgp0brDN6dtnXhSSUNP0fYd2dfimtNSNzLyYttIWMnBIVXLrscUQTtm3C/F3uP+PSmxiTIbzo9zGmnFY2IPKKRNoWNQ/Irmseh+jw+4O+AMgHBYbdnW6AixKt6owZjXHLu7l3hky2kLkzduZaMn8rfPC+id4nRYiJcdnYjqppy2Ya9xny/H75AWAJbyqlA0oKcDZLUqWoV8wOw/Ohizs306qb70Mg45yDP4Lim7t3Bx1HTiYxNMB8lMpib/idktgvMSBs4O2d2JxH11bFF5UucuDA14q3H0xg6oMdkLEHB7teFVjUS1Kac9mu1Z2GrZt3MIM4V9GruuXoIkPbro0E3NyGYCs757TsYNvMSAzACenPtcLSteJofLnNL3MHCT6vqEBREUYfudL0lSXk1QHPFfwhlS+GjAb+pdjEpKwh2JwH3C5IW/BloRbjzLKPK/pPJolhqpbYUffw2ggg13ZNPnVGUXWSR8jCRuS53cjQHqLieKzy0Z44tGODEGzT26IbV5oghEwQVcQfG9KbkojkZrzVWdydPXyS7I8cKIjocHQzNpelO2akr/vGgWuX6laUXggE2DJjDdBynxL5g+z1Lvh/XlJlCAwclaFP43gbJB2F5Bvux7mNe+fvAa0oXdDCjaKwD9sXZTf9VaZaNbVkWTPyfghKwrzshCLcyksYqLTYWhtWRr3fuhGBWRc7KPcjdTaptyJvkkJ7DN6lW9PNySR61tZqLgG4EzAXf2S8Kla9knuW9D0RtcJqsu44mvpwcAnfhuFBOVWN+7rZ8YeYwrGdpjsEdg=",IV=0x5886c1c418511d1f9fcc3b91f12b7e82
...
```
Any clue?
Thanks!

**Comment by bastimeyer** [comment, marked as answer]
> `#EXT-X-KEY:METHOD=AES-128,URI="chave://...`

That's a custom decryption key implementation by the host. This is unsupported by Streamlink (or any other 3rd party HLS client).

You'll have to write your own plugin for this and add a `python-requests` adapter for the custom `chave://` protocol, if you can figure out what the remaining key data means, which is base64 encoded. This could be the key itself, or something else required for retrieving the decryption key.

**Reply by bastimeyer** [reply]
Also see https://streamlink.github.io/cli.html#cmdoption-hls-segment-key-uri for overriding HLS decryption keys

**Reply by expend20** [reply]
Unfortunately base64 decoded data didn't make sense to me. 
**But!** Just passing proper `--hls-segment-key-uri` solved the problem for me!

**Comment by expend20** [comment]
The content is ecrypted with aes128, and the key is sent to the client as a response to subsequent GET request, just after m3u8 file.

**Reply by expend20** [reply]
thanks @bastimeyer !

**Reply by akagusu** [reply]
I have a video I want to download which have this exact same pattern then yours, but I followed the answer and I just go a 403 Forbidden while trying to download the segments.

Sorry for ask, but how did you do? Can you document what you did?

**Reply by expend20** [reply]
like this: `streamlink -o out.ts <url_with_m3u8_list> best --hls-segment-key-uri <url_to_aes128_bit_key>`. I think you should do it from exact the same IP and no longer than 2-3 minutes after you opened the page in your browser.

**Reply by akagusu** [reply]
I tried but I got the same error.

I've noticed that if I try to make the request to the key or to the m3u8 file in dev tools I got an "Access Denied" error. 

I don't know what is but I know that it is something beyond my knowledge.

Anyway, thank you.

**Reply by ghost** [reply]
Hello, can you help me?

I am facing same issue with this m3u8 stream:  https://dokko1new.koskoros.ru/dokko1/premium588/mono.m3u8

How do I find the URL that has the aes key from browser console?

**Comment by svenop5** [comment]
@expend20 I have a similar issue. I was able to successfully decrypt your encrypted key URI. It is encrypted with a key derived from a password embedded in the JavaScript. The decrypted key URI is the actual URI required to get the segment decryption key from the server.
